### PR TITLE
implement `ExactSizeIterator` for `slice::{Chunks,ChunksMut,Windows}`

### PR DESCRIPTION
**Pull request by apasel422**

### CODE REVIEW

File: src/libcore/slice.rs [hunk 1]
```diff
@@ -1212,6 +1212,9 @@ impl<'a, T> Iterator for Windows<'a, T> {
     }
 }
 
+#[stable]
+impl<'a, T> ExactSizeIterator for Windows<'a, T> {}
```
Comment (alexcrichton) [inline src/libcore/slice.rs]: This implementation may not forever be right as the above `size_hint` may sometimes return `None` for the upper bound (albeit somewhat rarely)

**Comment by alexcrichton** [comment]
Thanks! Could you add some tests for the new implementations as well?

File: src/libcore/slice.rs (hunk 1 above)
Comment (apasel422) [inline src/libcore/slice.rs]: I assume I should remove this impl, then?

Comment (alexcrichton) [inline src/libcore/slice.rs]: For now, yes, let's remove it until we're guaranteed that `size_hint` always returns an exact size hint.

**Comment by apasel422** [comment]
Updated.

**Comment by alexcrichton** [comment]
@bors: r+ 1a76263

Thanks!

**Comment by apasel422** [comment]
I had to update this due to missing imports and a mutability issue in the tests.

**Pull request merged**